### PR TITLE
add tke "extra_args"

### PR DESCRIPTION
**Pull request by ChuGuai**
terraform-provider-tencentcloud guojunchu$ make testacc TESTARGS="-run TestAccTencentCloudTkeAsResource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run TestAccTencentCloudTkeAsResource -timeout 120m
?       github.com/tencentcloudstack/terraform-provider-tencentcloud    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc     (cached) [no tests to run]
=== RUN   TestAccTencentCloudTkeAsResource
--- PASS: TestAccTencentCloudTkeAsResource (220.59s)
PASS
ok      github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud       225.508s
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity  [no test files]
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper       [no test files]
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit     [no test files]

terraform-provider-tencentcloud guojunchu$ make testacc TESTARGS="-run TestAccTencentCloudTkeScaleWorkerResource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run TestAccTencentCloudTkeScaleWorkerResource -timeout 120m
?       github.com/tencentcloudstack/terraform-provider-tencentcloud    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc     (cached) [no tests to run]
=== RUN   TestAccTencentCloudTkeScaleWorkerResource
--- PASS: TestAccTencentCloudTkeScaleWorkerResource (330.17s)
PASS
ok      github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud       334.818s
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity  [no test files]
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper       [no test files]
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit     [no test files]

terraform-provider-tencentcloud guojunchu$ make testacc TESTARGS="-run TestAccTencentCloudTkeResource"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run TestAccTencentCloudTkeResource -timeout 120m
?       github.com/tencentcloudstack/terraform-provider-tencentcloud    [no test files]
testing: warning: no tests to run
PASS
ok      github.com/tencentcloudstack/terraform-provider-tencentcloud/gendoc     (cached) [no tests to run]
=== RUN   TestAccTencentCloudTkeResource
--- PASS: TestAccTencentCloudTkeResource (239.11s)
PASS
ok      github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud       242.203s
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/connectivity  [no test files]
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper       [no test files]
?       github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/ratelimit     [no test files]

### CODE REVIEW

File: tencentcloud/resource_tc_kubernetes_scale_worker.go [hunk 1]
```diff
@@ -52,6 +100,8 @@ import (
 	"log"
 	"strings"
 
+	"github.com/tencentcloudstack/terraform-provider-tencentcloud/tencentcloud/internal/helper"
+
```
Comment (gailwang) [inline tencentcloud/resource_tc_kubernetes_scale_worker.go:104]: 去掉空行

File: tencentcloud/resource_tc_kubernetes_as_scaling_group_test.go [hunk 2]
```diff
@@ -128,6 +128,7 @@ resource "tencentcloud_kubernetes_cluster" "managed_cluster" {
   cluster_name            = "tf-tke-unit-test"
   cluster_desc            = "test cluster desc"
   cluster_max_service_num = 32
+  cluster_version		  = "1.18.4"
```
Comment (brickzzhang) [inline tencentcloud/resource_tc_kubernetes_as_scaling_group_test.go:131]: 格式化

File: tencentcloud/resource_tc_kubernetes_as_scaling_group_test.go [hunk 3]
```diff
@@ -206,6 +207,9 @@ resource "tencentcloud_kubernetes_as_scaling_group" "as_test" {
     "test1" = "test1",
     "test2" = "test2",
   }
+  extra_args = [
```
Comment (brickzzhang) [inline tencentcloud/resource_tc_kubernetes_as_scaling_group_test.go:210]: 增加属性之后在step中加下相关校验吧

File: tencentcloud/resource_tc_kubernetes_cluster_test.go [hunk 4]
```diff
@@ -160,6 +160,7 @@ resource "tencentcloud_kubernetes_cluster" "managed_cluster" {
   cluster_desc                               = "test cluster desc"
   cluster_max_service_num                    = 32
   cluster_internet                           = true
+  cluster_version		  					 = "1.18.4"
```
Comment (brickzzhang) [inline tencentcloud/resource_tc_kubernetes_cluster_test.go:163]: 格式化

File: tencentcloud/resource_tc_kubernetes_scale_worker_test.go [hunk 5]
```diff
@@ -153,6 +153,7 @@ resource "tencentcloud_kubernetes_cluster" "managed_cluster" {
   cluster_cidr            = "192.168.0.0/16"
   cluster_max_pod_num     = 32
   cluster_name            = "test"
+  cluster_version		  = "1.18.4"
```
Comment (brickzzhang) [inline tencentcloud/resource_tc_kubernetes_scale_worker_test.go:156]: 格式化

File: tencentcloud/service_tencentcloud_tke.go [hunk 6]
```diff
@@ -511,7 +518,7 @@ func (me *TkeService) DescribeClusterSecurity(ctx context.Context, id string) (r
 	return me.client.UseTkeClient().DescribeClusterSecurity(request)
 }
 
-func (me *TkeService) CreateClusterAsGroup(ctx context.Context, id, groupPara, configPara string, labels []*tke.Label) (asGroupId string, errRet error) {
+func (me *TkeService) CreateClusterAsGroup(ctx context.Context, id, groupPara, configPara string, labels []*tke.Label, iAdvanced InstanceAdvancedSettings) (asGroupId string, errRet error) {
```
Comment (brickzzhang) [inline tencentcloud/service_tencentcloud_tke.go:521]: 形参修改注意检查下存量代码的兼容性

File: tencentcloud/service_tencentcloud_tke.go [hunk 7]
```diff
@@ -524,6 +531,11 @@ func (me *TkeService) CreateClusterAsGroup(ctx context.Context, id, groupPara, c
 	request.ClusterId = &id
 	request.AutoScalingGroupPara = &groupPara
 	request.LaunchConfigurePara = &configPara
+	if len(iAdvanced.ExtraArgs.Kubelet) > 0 {
```
Comment (brickzzhang) [inline tencentcloud/service_tencentcloud_tke.go:534]: 注意取空地址时的异常

File: tencentcloud/resource_tc_kubernetes_cluster.go [hunk 8]
```diff
@@ -1,6 +1,9 @@
 /*
 Provide a resource to create a kubernetes cluster.
 
+~> **NOTE:** "extra_args" needs to be whitelist.
+~> **NOTE:** To use the custom Kubernetes component startup parameter function, you need to submit a ticket for application.
```
Comment (gailwang) [inline tencentcloud/resource_tc_kubernetes_cluster.go:5]: 合在一句就好了。。

File: tencentcloud/resource_tc_kubernetes_cluster.go [hunk 9]
```diff
@@ -1,6 +1,8 @@
 /*
 Provide a resource to create a kubernetes cluster.
 
+~> **NOTE:** To use the custom Kubernetes component startup parameter function, you need to submit a ticket for application.
```
Comment (gailwang) [inline tencentcloud/resource_tc_kubernetes_cluster.go:4]: **NOTE:** To use the custom Kubernetes component startup parameter function（parameter `extra_args`）, you need to submit a ticket for application.

**Review by gailwang — APPROVED** [review]

**Review by brickzzhang — APPROVED** [review]

**Review by oliverpei — APPROVED** [review]

**Pull request merged**